### PR TITLE
Tests: Allow Karma to load unminfied source

### PR DESCRIPTION
**Pull request by gibson042**
### Summary ###
Makes Karma behave more like the old tests by exposing a "load unminified" checkbox.

### Checklist ###
* [x] All authors have signed the CLA at https://cla.js.foundation/jquery/jquery
* [x] New tests have been added to show the fix or feature works
* [x] Grunt build and unit tests pass locally with these changes
* [x] ~~If needed, a docs issue/PR was created at https://github.com/jquery/api.jquery.com~~

### CODE REVIEW

**Comment by timmywil** [comment]
@gibson042 The `grunt karma:main` task seems to stall.

**Comment by gibson042** [comment]
@timmywil That was surprisingly tricky to address, but should be good to go now.

File: test/data/testinit.js [hunk 1]
```diff
@@ -271,12 +277,9 @@ this.testIframe = function( title, fileName, func, wrapper ) {
...
 
-if ( window.__karma__ ) {
-	// In Karma, files are served from /base
-	baseURL = "base/test/data/";
-} else {
-	// Tests are always loaded async
-	// except when running tests in Karma (See Gruntfile)
+// Tests are always loaded async
+// except when running tests in Karma (See Gruntfile)
+if ( !window.__karma__  ) {
```
Comment (mgol) [inline test/data/testinit.js]: A double space incorrectly got inserted after `__karma__`.

**Comment by gibson042** [comment]
Right, AMD behavior is at odds with the way the tests are currently loaded in Karma. I just removed that mode, because it doesn't provide much value over e.g. `grunt karma:chrome-debug` anyway.

File: test/data/testinit.js (hunk 1 above)
Comment (gibson042) [reply test/data/testinit.js]: Fixed.

**Comment by mgol** [comment]
> I just removed that mode, because it doesn't provide much value over e.g. `grunt karma:chrome-debug` anyway.

What do you mean by that?

**Comment by gibson042** [comment]
I typed the wrong example command, and looking back it wasn't really coherent anyway. Please disregard.

However, we still don't have AMD mode in Karma, and I'm disinclined to force that in scope for this PR (there's no loss, since current master lacks both AMD and unminified). Someone else can make those changes later if they so desire.

**Comment by mgol** [comment]
I agree, adding AMD support is a separate feature request.

**Pull request merged**